### PR TITLE
Fixes #33759 - Deliver cancellation notifications to clients

### PR DESCRIPTION
**Pull request by adamruzicka**
over MQTT

### CODE REVIEW

File: lib/smart_proxy_remote_execution_ssh/actions/pull_script.rb [hunk 1]
```diff
@@ -81,6 +80,21 @@ def mqtt_start(otp_password)
...
     end
 
+    def mqtt_cancel
+      cleanup
+      payload = {
+        type: 'data',
+        message_id: SecureRandom.uuid,
+        version: 1,
+        sent: DateTime.now.iso8601,
+        directive: 'cancel',
```
Comment (ezr-ondrej) [inline lib/smart_proxy_remote_execution_ssh/actions/pull_script.rb:90]: The directive needs to be `foreman`, that informs the yggdrassil what worker to deliver this to.
I guess we will have to put the cacel in the metadata in some way

**Comment by adamruzicka** [comment]
@ezr-ondrej ping

**Comment by ezr-ondrej** [comment]
Looks good, but needs to have the worker counterpart, I'll try to work on that asap.

**Review by ezr-ondrej — APPROVED** [review]
LGTM :+1:

**Comment by adamruzicka** [comment]
Thank you @ezr-ondrej !

**Pull request merged**